SimTKcommon/Geometry/src/DecorativeGeometryRep.h

```diff
@@ -583,6 +583,174 @@ const DecorativeMeshFile& getMyMeshFileHandle() const {
+public:
+    // no default constructor
+    explicit DecorativeTorusRep(Real torusR, Real tubeR) : torusRadius(torusR), tubeRadius(tubeR) {
+    }
+
+    const Real& getTorusRadius() const {
+        return  torusRadius;
+    }
+
+    const Real& getTubeRadius() const {
+        return  tubeRadius;
+    }
+
+    void setTorusRadius(Real torusR) {
+        torusRadius = torusR;
+    }
+
+    void setTubeRadius(Real tubeR) {
+        tubeRadius = tubeR;
+    }
+    // virtuals
+    DecorativeGeometryRep* cloneDecorativeGeometryRep() const {
+        DecorativeTorusRep* DGRep = new DecorativeTorusRep(*this);
+        return DGRep;
+    }
+
+    void implementGeometry(DecorativeGeometryImplementation& geometry) const {
+        geometry.implementTorusGeometry(getMyTorusHandle());
+    }
+
+    SimTK_DOWNCAST(DecorativeTorusRep, DecorativeGeometryRep);
+private:
+    Real torusRadius;
+    Real tubeRadius;
+
+    // This is just a static downcast since the DecorativeGeometry handle class is not virtual.
+
+    const DecorativeTorus& getMyTorusHandle() const {
+        return *static_cast<const DecorativeTorus*>(myHandle);
+    }
+};
+
+
+///////////////////////
+// DecorativArrowRep //
+///////////////////////
+
+class DecorativeArrowRep : public DecorativeGeometryRep {
+public:
+    // no default constructor
+    explicit DecorativeArrowRep(const Vec3& start, const Vec3& end, Real& tipLen) : 
+        start_point(start), end_point(end), tip_length(tipLen) {
+    }
+
+    const Vec3& getStartPoint() const {
+        return  start_point;
+    }
+
+    const Vec3& getEndPoint() const {
+        return  end_point;
+    }
+
+    const Real& getTipLength() const {
+        return  tip_length;
+    }
+
+    void setStartPoint(const Vec3& start) {
+        start_point = start;
+    }
+    void setEndPoint(const Vec3& end) {
+        end_point = end;
+    }
+    void setTipLength(Real& tlen) {
+        tip_length = tlen;
+    }
+
+    // virtuals
+    DecorativeGeometryRep* cloneDecorativeGeometryRep() const {
+        DecorativeArrowRep* DGRep = new DecorativeArrowRep(*this);
+        return DGRep;
+    }
+
+    void implementGeometry(DecorativeGeometryImplementation& geometry) const {
+        geometry.implementArrowGeometry(getMyArrowHandle());
+    }
+
+    SimTK_DOWNCAST(DecorativeArrowRep, DecorativeGeometryRep);
+private:
+    Vec3 start_point;
+    Vec3 end_point;
+    Real tip_length;
+
+    // This is just a static downcast since the DecorativeGeometry handle class is not virtual.
+
+    const DecorativeArrow& getMyArrowHandle() const {
+        return *static_cast<const DecorativeArrow*>(myHandle);
+    }
+};
+
+
+
+///////////////////////
+// DecorativeConeRep //
+///////////////////////
+
+class DecorativeConeRep : public DecorativeGeometryRep {
+public:
+    // no default constructor
+    explicit DecorativeConeRep(const Vec3& orig, const Vec3& dir, Real& h, Real& baseR) : origin(orig), direction(dir), height(h), baseRadius(baseR) {
```

Comment:
> Any preference as to whether the colon should go with the top or bottom line

Yes, I prefer the colon on the bottom line where I think it is more visible. And I like to align the first initializer at the first tab stop as though it were code since initializers act like assignments. Then the whole thing looks like
```c++
explicit DecorativeConeRep(const Vec3& orig, const Vec3& dir, Real& h, Real& baseR) 
:   origin(orig), direction(dir), height(h), baseRadius(baseR)  {
    int localVar = 0;
    for (;;) {
        // some code
    }
}
```

This is a mild preference; there are lots of other ways to squish the code into 80 characters that are still readable. I like to avoid stretching the code out vertically and the above format does a pretty good job of being compact but clear (IMO).